pkg/appliance/upgrade.go

```diff
@@ -0,0 +1,504 @@
+package appliance
```

Comment:
This is where the upgrade complete logic is handled now, along with checks and batching.